crates/sdk/src/queries/shell.rs

```diff
@@ -131,6 +138,36 @@ where
     unimplemented!("Dry running tx requires \"wasm-runtime\" feature.")
 }
 
+/// Return an estimate of the maximum time taken to decide a block
+fn max_block_time<D, H, V, T>(
+    ctx: RequestCtx<'_, D, H, V, T>,
+) -> namada_storage::Result<DurationSecs>
+where
+    D: 'static + DB + for<'iter> DBIter<'iter> + Sync,
+    H: 'static + StorageHasher + Sync,
+{
+    // NB: get max time over this num of blocks
+    const NUM_BLOCKS_TO_READ: u64 = 5;
```

Comment:
Should we allow the caller to pass this value as an argument?

Reply:
no, because this is an unprivileged rpc method. a value too high can cause nodes to block for too long performing storage reads